lib/active_publisher/logging.rb

```diff
@@ -0,0 +1,27 @@
+# Taken from https://github.com/mperham/sidekiq/blob/7f882787e53d234042ff18099241403300a47585/lib/sidekiq/logging.rb
+require "time"
+require "logger"
+
+module ActivePublisher
+  module Logging
+    def self.initialize_logger(log_target = STDOUT)
+      oldlogger = defined?(@logger) ? @logger : nil
+      @logger = Logger.new(log_target)
+      @logger.level = Logger::INFO
+      oldlogger.close if oldlogger && !$TESTING # don't want to close testing's STDOUT logging
+      @logger
+    end
+
+    def self.logger
+      defined?(@logger) ? @logger : initialize_logger
+    end
+
+    def self.logger=(log)
+      @logger = (log ? log : Logger.new("/dev/null"))
```

Comment:
I know a lot of the considerations around this are for allowing us to set `logger = nil` and have it still work. We actually lifted this code from the `sidekiq` project so I don't know if we have a lot of understanding around all of the nuances ourselves. We were trying to leverage the hard-won experience of the `sidekiq` project without having to learn it all ourselves.